### PR TITLE
Read worker http ports from etcd

### PR DESCRIPTION
**Pull request by ChunxuTang**
This PR reads workers' http ports from the etcd. So workers could have different ports.

Waiting for a release of Alluxio which includes the commit: [262b2b339a1779b2085d2c73654d3d21ec6ffa33](https://github.com/Alluxio/alluxio/pull/18499)

### CODE REVIEW

File: alluxio/worker_ring.py [hunk 1]
```diff
@@ -77,23 +81,29 @@ def from_worker_info(worker_info):
...
+            "WorkerNetAddress{{host={}, containerHost={}, rpcPort={}, "
+            "dataPort={}, webPort={}, domainSocketPath={}, httpServerPort={}}}"
         ).format(
             self.host,
             self.container_host,
             self.rpc_port,
             self.data_port,
             self.web_port,
             self.domain_socket_path,
+            self.http_server_port,
```
Comment (LuQQiu) [inline alluxio/worker_ring.py:106]: worker_string = worker_address.dump_main_info()
                for i in range(weight):
                    hash_key = self._hash(worker_string, i)
                    hash_ring[hash_key] = worker_address
                    
dump_main_info is used to construct the hash ring and you are changing the hash ring which may make all "loaded files" useless

Comment (LuQQiu) [reply alluxio/worker_ring.py:106]: and this suppose to be same as the Alluxio server side which i am not quite sure whether they have changed on Alluxio server side or not

Comment (jja725) [reply alluxio/worker_ring.py]: I think Lu means we need http port in dump_main_info

**Review by jja725 — COMMENTED** [review]
Beside Lu's comment, I think it would be better to have dataclass similar to worker_net_address instead of having separate parameters for each function

**Comment by ChunxuTang** [comment]
@LuQQiu @jja725 Per our offline discussion, I'll merge this PR first for the following testing.
There will be follow-up PRs for the hash ring construction.

**Pull request merged**